test/test_arhp.rb

```diff
@@ -55,6 +55,22 @@ def test_should_insert_on_correct_database
     assert_action_uses_correct_database(:insert, "insert into tests values(NULL, 'foo')")
   end
 
+  def test_models_with_matching_hosts_and_non_matching_databases_should_share_a_connection
+    simulate_rails_app_active_record_railties
+    assert_equal(Test1.connection.raw_connection, Test1Shard.connection.raw_connection)
```

Comment:
Could we just use `Test2` here and below, removing the need for `Test1Shard` and the configuration?

I think we should still keep your comment though, e.g.

```ruby
# test/helper.rb

    eval <<-RUBY
      # The placement of these classes is important. The
      # most recent connection established should not be
      # for test_host_1. The following test relies on it:
      # test_models_with_matching_hosts_and_non_matching_databases_do_not_mix_up_underlying_database
      class Test1 < ActiveRecord::Base
```

Reply:
We can, but it will require updating [this test](https://github.com/zendesk/active_record_host_pool/blob/fd17cfdbeb16ec296ad335a19fe35b20486e1ac5/test/test_arhp.rb#L54-L56) because right now there is an assumption in the test suite that the pre-existing `TestN` models all have the "tests" table in their database.

For our test we need a table that is one database but not the other which helps recreate the observable issue.

WDYT?